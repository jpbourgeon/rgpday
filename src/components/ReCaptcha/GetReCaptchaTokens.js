import React from 'react'
import ReCaptcha from './'

class GetReCaptchaTokens extends React.Component {
  constructor (props) {
    super(props)
    this.state = { token: '' }
    this.ReCaptchaRef = React.createRef()
    this.getNewRecaptcha = this.getNewRecaptcha.bind(this)
    this.handleReCaptchaToken = this.handleReCaptchaToken.bind(this)
  }

  getNewRecaptcha () {
    this.ReCaptchaRef.current.getToken()
  }

  handleReCaptchaToken (token) {
    this.setState({ token })
  }

  render () {
    return (
      <div style={{ margin: '1em' }}>
        <code>
          <h1>rgpday.com reCaptcha</h1>
          <div><button onClick={this.getNewRecaptcha}>Get a token</button>
          </div><br />
          <div>{this.state.token}</div>
          <ReCaptcha
            action='contact'
            sitekey='6LedLpMUAAAAAG8Ai4M4x9wTcIs4rPmvYV82a7Yh'
            verifyCallback={this.handleReCaptchaToken}
            ref={this.ReCaptchaRef}
          />
        </code>
      </div>
    )
  }
}

export default GetReCaptchaTokens
