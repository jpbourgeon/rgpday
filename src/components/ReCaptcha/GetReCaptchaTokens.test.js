/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import GetReCaptchaTokens from './GetReCaptchaTokens'
import jssSerializer from '../../utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The GetReCaptchaTokens component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<GetReCaptchaTokens />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
