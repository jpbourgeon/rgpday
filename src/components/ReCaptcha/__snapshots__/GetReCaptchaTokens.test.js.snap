// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The GetReCaptchaTokens component should render correctly 1`] = `
<div
  style={
    Object {
      "margin": "1em",
    }
  }
>
  <code>
    <h1>
      rgpday.com reCaptcha
    </h1>
    <div>
      <button
        onClick={[Function]}
      >
        Get a token
      </button>
    </div>
    <br />
    <div>
      
    </div>
  </code>
</div>
`;
